### ISSUE TITLE
Memory management 

### ISSUE THREAD

**Issue by JenishKananiLime** [mapbox/mapbox-maps-ios#1534]
## Environment
How can we clean up memory?

for example before I add all annotations and styles the memory is around 250 mb after which it goes to 400+ mb questions is that if we remove this style and annotations meomry goes to 350 to 400 mb is there a way to reclaim more memory so app doesn’t crash later?
- Xcode version:
- iOS version:
- Devices affected:
- Maps SDK Version:

**Comment by ZiZasaurus** [comment]
@JenishKananiLime how are you removing annotations from your map? Are you using the method outlined [here](https://docs.mapbox.com/ios/maps/guides/annotations/annotations/#removing-annotations)?

**Comment by JenishKananiLime** [comment]
Yes that is what I do, but evening removing all annotations / styles the memory still doesn't go down as much
